language-design.md

```diff
@@ -17,7 +17,24 @@ For each quote, describe what it was about the quote that led you pick it.
 
 **Response**
 
+>Names matter. Strive for intelligibility, consistency, and
+>symmetry. Every API is a little language, and people must learn to
+>read and write it. If you get an API right, code will read like
+>prose. [Bloch, 2006]
 
+Both Alex and Zoab felt like this point resonated with them. Especially when reading someone else's code, Zoab has observed that the quality of names affects the ease with which he can understand the code - variable names, method names, etc. that correspond to the semantics of the construct allow you to get an idea of how the construct works without having to closely read the implementation. Alex says "Every name is like a gateway. If you understand the name, you can stop reading. If you don't understand, you have to then go read all of the implementation behind that name."
+
+They both recognize that choosing good names can be a struggle, especially when dealing with abstract or complex constructs, but they both think it's worth putting in the effort. There will always be people who misunderstand your naming convention, but if you put effort into choosing names, you can minimize that set. 
+
+> This leads me to claim that, from now on, a main goal in designing a language should be to plan for growth. [Steele, 1998]
+
+Alex and Zoab don't necessarily agree with this statement. They understand that the person making this statement is a general purpose programming language designer, and they can understand the validity of this claim for general purpose programming languages. However, it's not obvious to them that for DSLs in particular, you'd want to design them with a plan for growth. Consider a language that has already expanded to fill its domain, where its domain is likely to remain static. In this case it wouldn't necessarily be a good idea to plan for growth. An example of this kind of language is Picobot - if there's a trade off between making Picobot easier to use or making it easier to grow, we should choose to make it easy to use because we don't expect to have to grow it. 
```

Comment:
Anna thinks that this response is true if growth always leads to expanding the domain, but "growth" and "domain" may be ambiguous enough that it may be possible to grow without expanding the domain. Daniel thinks that examples like Picobot are relatively rare; Picobot is designed not to write anything but to teach things, while most languages are designed to help solve some problem. We also think that sometimes DSLs are connected to general purpose programming languages or other tools, and may need to change with those tools even if the domain remains the same.